crates/gen-host-js/src/lib.rs

```diff
@@ -875,11 +964,17 @@ impl Instantiator<'_> {
         iface: Option<&Interface>,
     ) {
         if exports.is_empty() {
-            self.src.js("{}");
+            if self.gen.opts.instantiation {
+                self.src.js("{}");
+            }
```

Comment:
Personally I would ideally prefer to have sort of a mish-mash sprinkling of `if self.gen.opts.instantiation` throughout since I feel that makes code difficult to read and change in the future. Would it be possible to instead change the return value of this function to something that works in both `--instantiation` and not modes? For example the `exports` here could return a form of `HashMap<???, String>` or something and then the caller would, once, check `instantiation` and create an export object or create a bunch of exported functions.

Reply:
By switching to namespaces there is now much more consistent handling between the modes. Overall I think this gating is a semantically important one as it represents the difference between exporting a namespace and a top-level declaration.